src/Analysis/Ast/Impl/Indexing/Symbols.cs

```diff
@@ -70,6 +72,8 @@ internal class HierarchicalSymbol {
             Children = children;
             _functionKind = functionKind;
         }
+
+        public IEnumerable<(HierarchicalSymbol, string)> ChildrenWithParentName() => Children.MaybeEnumerate().Select((child) => (child, Name));
```

Comment:
This function is specific to the way that the indexer walks the tree during a query. I think this should be a function inside of `SymbolIndex` instead (which would keep this class as purely data).